From 1c8538d8f8c2181106d626d67784af6db094036e Mon Sep 17 00:00:00 2001
From: Rafostar <rafostar.github@gmail.com>
Date: Thu, 19 Nov 2020 18:03:11 +0100
Subject: [PATCH] assrender: fix smooth scaling by disabling hinting

When ass hinting value is set to anything other than NONE,
subtitles cannot use smooth scaling, thus all animations will jitter.

The libass author warns about possibility of breaking some scripts when it is enabled,
so lets do what is recommended and disable it to get the smooth scaling working.
---
 ext/assrender/gstassrender.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/assrender/gstassrender.c b/ext/assrender/gstassrender.c
index e99458bf29..111987b9d8 100644
--- a/ext/assrender/gstassrender.c
+++ b/ext/assrender/gstassrender.c
@@ -916,7 +916,7 @@ gst_ass_render_negotiate (GstAssRender * render, GstCaps * caps)
     ass_set_pixel_aspect (render->ass_renderer,
         (gdouble) render->info.par_n / (gdouble) render->info.par_d);
     ass_set_font_scale (render->ass_renderer, 1.0);
-    ass_set_hinting (render->ass_renderer, ASS_HINTING_LIGHT);
+    ass_set_hinting (render->ass_renderer, ASS_HINTING_NONE);
 
     ass_set_fonts (render->ass_renderer, "Arial", "sans-serif", 1, NULL, 1);
     ass_set_fonts (render->ass_renderer, NULL, "Sans", 1, NULL, 1);
-- 
GitLab

